### PR TITLE
SoundPlayer: Port to LibMain

### PR DESCRIPTION
**Pull request by elyse0**
Use LibMain and TRY in SoundPlayer. (:

### CODE REVIEW

File: Userland/Applications/SoundPlayer/main.cpp [hunk 1]
```diff
@@ -19,118 +20,113 @@
...
     window->set_title("Sound Player");
     window->set_icon(app_icon.bitmap_for_size(16));
 
-    String path = argv[1];
+    String path = arguments.argv[1];
     // start in advanced view by default
-    Player* player = &window->set_main_widget<SoundPlayerWidgetAdvancedView>(window, audio_client);
-    if (argc > 1) {
+    Player* player = TRY(window->try_set_main_widget<SoundPlayerWidgetAdvancedView>(window, audio_client));
+    if (arguments.argc > 1) {
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/SoundPlayer/main.cpp:44]: Accessing the argument before ensuring it exists is a silent index-out-of-bounds and could be easily drive-by-fixed:
```suggestion
    // start in advanced view by default
    Player* player = TRY(window->try_set_main_widget<SoundPlayerWidgetAdvancedView>(window, audio_client));
    if (arguments.argc > 1) {
        String path = arguments.strings[1];
```

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]

File: Userland/Applications/SoundPlayer/main.cpp (hunk 1 above)
Comment (elyse0) [reply Userland/Applications/SoundPlayer/main.cpp]: Thanks, nice catch! 🐞
